src/main/java/racingcar/controller/RacingGameController.java

```diff
@@ -1,28 +1,36 @@
 package racingcar.controller;
 
 import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.GetMapping;
 import org.springframework.web.bind.annotation.PostMapping;
 import org.springframework.web.bind.annotation.RequestBody;
 import org.springframework.web.bind.annotation.RestController;
-
-import racingcar.controller.dto.RacingInfoRequest;
-import racingcar.controller.dto.RacingInfoResponse;
-import racingcar.domain.CarGroup;
+import racingcar.controller.dto.RacingGameRequest;
+import racingcar.controller.dto.RacingGameResponse;
+import racingcar.domain.Cars;
 import racingcar.service.RacingGameService;
 
+import java.util.List;
+
 @RestController
 public class RacingGameController {
-
-    private final RacingGameService racingGameServiceImpl;
+    private final RacingGameService racingGameService;
 
     public RacingGameController(final RacingGameService racingGameService) {
-        this.racingGameServiceImpl = racingGameService;
+        this.racingGameService = racingGameService;
     }
 
     @PostMapping("/plays")
-    public ResponseEntity<RacingInfoResponse> playRacingGame(@RequestBody RacingInfoRequest request) {
-        final CarGroup carGroup = new CarGroup(request.getNames());
-        final RacingInfoResponse response = racingGameServiceImpl.race(carGroup, request.getCount());
+    public ResponseEntity<RacingGameResponse> playRacingGame(@RequestBody RacingGameRequest request) {
+        final Cars cars = new Cars(request.getNames());
```

Comment:
찰리가 말씀해주신대로 Controller에 도메인 객체가 나오면 안된다고 생각해서 수정했습니다.
완성된 객체를 Service에 넣어서 로직을 실행시킨다는 생각에 Controller에서 생성시켰습니다.

이럴 경우 도메인 객체를 Controller에서 실행시킨다거나 혹은 중요한 정보를 응답에 담아 내보낼 수도 있는 위험성이 있으니 주의하겠습니다 !

Reply:
미리 말씀드리지 못했는데 저는 개인적인 취향의 영역이라고는 생각합니다 :)

Service 에서 도메인 객체를 인자로 받아 사용한다면 다른 Service가 해당 Service 를 의존해야할 때 사용하기 편리할 거에요
이럴 때 아래와 같은것들을 고민해보면 좋겠어요
- Service 가 같은 계층(Layer)에 있는 Service 를 의존해도 될까?
- Service 끼리 계층이 명확하다면 서로 의존해도 되는것 아닐까?
- Service 끼리 계층 표현을 어떻게 하지?
- 계층 표현이 불가능하면 어떻게 하지?

헤나만의 정답을 찾아봤으면 좋겠어요 :)
크루들과 토론해보기도 좋은 주제라 생각합니다

Reply:
## Service가 Service를 의존할 경우 순환 참조 문제가 발생할 수 있다.

빈을 등록 시키기 위해서 
ServiceA는 ServiceB를 받으려하고
ServiceB는 ServiceA를 받으려하는 문제로
서로 계속 호출하는 경우입니다.

아래와 같이 컴파일 시점에서 순환 참조에 대해서 에러를 발생시킵니다.

```java
┌─────┐
|  serviceA defined in file []
↑     ↓
|  serviceB defined in file []
└─────┘

Relying upon circular references is discouraged and they are prohibited by default. Update your application to remove the dependency cycle between beans. As a last resort, it may be possible to break the cycle automatically by setting spring.main.allow-circular-references to true.
```
```java
public class ServiceA {
    private final ServiceB serviceB;
}

public class ServiceB {
    private final ServiceA serviceA;
}
```

## Service 끼리 계층이 명확할 경우 의존해도 괜찮을까 ?

저는 Service 계층이 명확하다는 뜻이 Service와 Repository가 1:1인 경우라고 생각했습니다.

그리고 다음과 같이 Service가 Service를 서로 의존하는 상황을 가정했습니다.

```java
@Service
public class UserService {
    private final PostService postService;
    private final UserRepository userRepository;

    public UserService(PostService postService, UserRepository userRepository) {
        this.postService = postService;
        this.userRepository = userRepository;
    }

    // PostService를 이용해서 사용자의 게시글을 전체 조회한다.
    public List<Post> findAllPostByUserId(long userId) {
        return postService.findAllByUserId(userId);
    }

    // PostService를 이용해서 사용자 정보와 함께 게시글을 저장한다.
    public long savePost(String title, String content, long userId) {
        User findUser = userRepository.findByUserId(userId);
        Post newPost = new Post(title, content, findUser.findUsername(), findUser.getId());
        return postService.save(newPost);
    }
}
```

- PostService를 이용해서 사용자의 게시글을 전체 조회한다.
- PostService를 이용해서 사용자 정보와 함께 게시글을 저장한다.

이런 식으로 PostService에 구현된 메서드를 재사용해서 중복되는 로직을 제거하는 경우 괜찮을 수도 있다 생각듭니다.
- UserService가 의존하는 PostService의 조회 기능만을 사용할 경우
- 순환 참조 문제가 일어나지 않는 경우

하지만 여러 단점이 보여서 저는 선호하지 않습니다.
- UserService가 다른 Service 의존도가 높아집니다.
- 순환 참조가 발생할 수 있습니다.
- UserService를 이용하기 위해서는 PostService 내부 로직을 알아야합니다.
- 트랙잭션 전파가 일어날 수 있어 복잡해질 수 있습니다.

## 계층 표현이 불가능하다면

계층 표현이 불가능한 Service 끼리 의존하게 된다면 복잡해질 가능성이 큽니다.

계층 표현이 가능하면 현재 Service가 의존하는 Service가 어떤 영향을 미칠지 알기 쉬습니다.
하지만 계층 표현이 불가능한 Service라면 어떤 영향을 미칠지 알기 힘들어집니다.